pyrometheus/__init__.py

```diff
@@ -384,40 +431,53 @@ def get_mix_molecular_weight(self, Y):
         return 1/np.dot( self.iwts, Y )
 
     def get_concentrations(self, rho, Y):
-        return self.iwts * rho * Y
-
-    def get_mixture_specific_heat_cp_mass(self, T, Y):
-        return self.gas_constant * np.sum(
-            self.get_species_specific_heats_R(T)* Y * self.iwts)
-
-    def get_mixture_specific_heat_cv_mass(self, T, Y):
-        cp0_R = self.get_species_specific_heats_R( T ) - 1.0
-        return self.gas_constant * np.sum(Y * cp0_R * self.iwts)
-
-    def get_mixture_enthalpy_mass(self, T, Y):
-        h0_RT = self.get_species_enthalpies_RT( T )
-        return self.gas_constant * T * np.sum(Y * h0_RT * self.iwts)
-
-    def get_mixture_internal_energy_mass(self, T, Y):
-        e0_RT = self.get_species_enthalpies_RT( T ) - 1.0
-        return self.gas_constant * T * np.sum(Y * e0_RT * self.iwts)
+        concs = self.iwts * rho * Y
+        zero = 0 * concs[0]
+        for i, conc in enumerate(concs):
+            concs[i] = self.usr_np.where(concs[i] > 0, concs[i], zero)
+        return concs
```

Comment:
That does not work as intended:
```py
E   ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

Reply:
Ah, so I still need to iterate through the array to do that?   If this expression is correct, then this "works" for `pyrometheus` proper - but unsure if it will continue working with the array context.

```
for (...):
   concs[i] = max(concs[i], 0)
```

Reply:
Turns out, didn't work.  If this change is made, the code stops getting the right answers on all the points in the MIRGE-Com context as tested [here](https://github.com/illinois-ceesd/mirgecom/pull/147).